zhaquirks/sonoff/swv.py

```diff
@@ -0,0 +1,36 @@
+"""Sonoff SWV - Zigbee smart water valve."""
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import QuirkBuilder
+import zigpy.types as t
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+
+class ValveState(t.enum8):
+    """Water valve state."""
+
+    Normal = 0
+    Water_Shortage = 1
+    Water_Leakage = 2
+    Water_Shortage_And_Leakage = 3
+
+
+class EwelinkCluster(CustomCluster):
+    """Ewelink specific cluster."""
+
+    cluster_id = 0xFC11
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Attribute definitions."""
+
+        water_valve_state = ZCLAttributeDef(
+            id=0x500C,
+            type=ValveState,
+        )
+
+    @property
+    def _is_manuf_specific(self):
+        return False
+
+
+(QuirkBuilder("SONOFF", "SWV").replaces(EwelinkCluster).add_to_registry())
```

Comment:
Pre-commit (ruff) enforces this formatting now, since the v2 quirk is a lot shorter with the enum sensor now..
Let's try this:
```suggestion
(
    QuirkBuilder("SONOFF", "SWV")
    .replaces(EwelinkCluster)
    .add_to_registry()
)  # fmt: skip
```